grpc/src/main/java/io/stargate/grpc/codec/cql/MapCodec.java

```diff
@@ -0,0 +1,76 @@
+package io.stargate.grpc.codec.cql;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.stargate.db.schema.Column.ColumnType;
+import io.stargate.proto.QueryOuterClass.Collection;
+import io.stargate.proto.QueryOuterClass.Value;
+import java.nio.ByteBuffer;
+
+public class MapCodec extends CompositeCodec {
+
+  @Override
+  public ByteBuffer encode(@NonNull Value value, @NonNull ColumnType type) {
+    if (!value.hasCollection()) {
+      throw new IllegalArgumentException("Expected collection type");
+    }
+    assert type.isMap();
+
+    Collection map = value.getCollection();
+
+    ColumnType keyType = type.parameters().get(0);
+    ValueCodec keyCodec = ValueCodecs.get(keyType.rawType());
+
+    ColumnType valueType = type.parameters().get(1);
+    ValueCodec valueCodec = ValueCodecs.get(valueType.rawType());
+
+    int elementCount = map.getElementsCount();
+    if (elementCount % 2 != 0) {
+      throw new IllegalArgumentException(
+          "Missing pair value (expected an even number of elements)");
```

Comment:
nit: how do we know it's the pair _value_ that is missing and not the key? :)

Reply:
Am I missing something? Keys are always odd.  `<key> (1), <value> (2), <key> (3), <value> (4) ...`.

Reply:
Sorry, I did not mean to cause confusion. What I mean is that since the array length is odd, something is mixed up, but can we be certain it's the "value" that is missing? The client might have submitted something completely irrelevant here, so it might be best to return `expected an even number of elements` without trying to interpret the contents. WDYT?

Reply:
Fixed this.